### DISCUSSION TITLE
How to redirect output of rich progress bar to file?

### DISCUSSION THREAD

**Discussion by ZeguanXiao** [Lightning-AI/pytorch-lightning#13229] [Lightning Trainer API: Trainer, LightningModule, LightningDataModule] [answered]
When using rich progress bar with shell command like _python train.py > file.log 2>&1_, the file.log don't save any progress bar output. But check rich library [documents](https://rich.readthedocs.io/en/latest/console.html#error-console), it says rich will write to sys.stdout by default. Anyone know how to do that?

**Comment by tshu-w** [comment, marked as answer]
In fact, it does redirect to file.log and you can see them if you wait until all is over. I think maybe you can submit an issue and see if PL can keep flushing when running.
